mlem/contrib/tensorflow.py

```diff
@@ -0,0 +1,216 @@
+        try:
+            ret = tf.convert_to_tensor(obj, dtype=getattr(tf, self.dtype))
+        except (ValueError, TypeError):
+            raise DeserializationError(  # pylint: disable=raise-missing-from
+                f"given object: {obj} could not be converted to tensor "
+                f"of type: {getattr(tf, self.dtype)}"
+            )
+        self._check_shape(ret, DeserializationError)
+        return ret
+
+    def get_requirements(self) -> Requirements:
+        return Requirements.new([InstallableRequirement.from_module(tf)])
+
+    def get_writer(
+        self, project: str = None, filename: str = None, **kwargs
+    ) -> DataWriter:
+        return TFTensorWriter(**kwargs)
+
+    def get_model(self, prefix: str = ""):
+        raise NotImplementedError
+
+    @classmethod
+    def process(cls, obj: tf.Tensor, **kwargs) -> DataType:
+        return TFTensorDataType(
+            shape=tuple(obj.shape),
+            dtype=obj.dtype.name,
+        )
+
+
+DATA_KEY = "data"
+
+
+class TFTensorWriter(DataWriter):
+    type: ClassVar[str] = "tensorflow"
+
+    def write(
+        self, data: DataType, storage: Storage, path: str
+    ) -> Tuple[DataReader, Artifacts]:
+        with storage.open(path) as (f, art):
+            np.savez_compressed(f, **{DATA_KEY: data.data.numpy()})
+        return TFTensorReader(data_type=data), {self.art_name: art}
+
+
+class TFTensorReader(DataReader):
+    type: ClassVar[str] = "tensorflow"
+
+    def read(self, artifacts: Artifacts) -> DataType:
+        if DataWriter.art_name not in artifacts:
+            raise ValueError(
+                f"Wrong artifacts {artifacts}: should be one {DataWriter.art_name} file"
+            )
+        with artifacts[DataWriter.art_name].open() as f:
+            np_data = np.load(f)[DATA_KEY]
+            data = tf.convert_to_tensor(
+                np_data, dtype=getattr(tf, np_data.dtype.name)
+            )
+            return self.data_type.copy().bind(data)
+
+    def read_batch(
+        self, artifacts: Artifacts, batch_size: int
+    ) -> Iterator[DataType]:
+        raise NotImplementedError
+
+
+def is_custom_net(model):
+    return (
+        not model._is_graph_network  # pylint:disable=protected-access
+        and not isinstance(model, sequential.Sequential)
+    )
+
+
+class TFKerasModelIO(BufferModelIO):
+    """
+    :class:`.ModelIO` implementation for Tensorflow Keras models (:class:`tensorflow.keras.Model` objects)
+    """
+
+    type: ClassVar[str] = "tensorflow_io"
+    save_format: Optional[str] = None
+
+    def save_model(self, model: tf.keras.Model, path: str):
+        if self.save_format is None:
+            self.save_format = "tf" if is_custom_net(model) else "h5"
+        model.save(path, save_format=self.save_format)
+
+    def load(  # pylint:disable=inconsistent-return-statements
+        self, artifacts: Artifacts
+    ):
+        if self.save_format == "h5":
+            if len(artifacts) != 1:
```

Comment:
better to check `self.art_name in artifacts`

Reply:
done